lib/iris/cube.py

```diff
@@ -2278,8 +2282,68 @@ def _intersect_modulus(self, coord, minimum, maximum, min_inclusive,
+                # coordinate.
+
+                # The following condition holds for increasing or decreasing
+                # coordinates with wraparound.
+                # Condition1: The two blocks don't themselves wrap
+                #             (inside_indices is contiguous).
+                # Condition2: Are we chunked at either extreme edge.
+                index_of_second_chunk = inside_indices[end_of_first_chunk + 1]
+                final_index = points.size - 1
+                if ((index_of_second_chunk ==
+                        inside_indices[end_of_first_chunk] + 1) and
+                        index_of_second_chunk in (final_index, 1)):
+
+                    # Check if final index is wrapped (increasing coordinate).
+                    # i.e. start of the second chunk is at the final index.
+                    if index_of_second_chunk == final_index:
+                        # Unwrap points and bounds
+                        if (bounds is not None and
+                                np.isclose(coord.bounds[-1, -1],
+                                           coord.bounds[0, 0] + modulus)):
+                            bounds[-1, :] = coord.bounds[-1, :]
+                            points[-1] = coord.points[-1]
+                            subsets = [slice(inside_indices[0],
+                                             inside_indices[-1] + 1)]
+                        # Unwrap points
+                        elif np.isclose(coord.points[-1],
+                                        coord.points[0] + modulus):
+                            points[-1] = coord.points[-1]
+                            subsets = [slice(inside_indices[0],
+                                       inside_indices[-1] + 1)]
+
+                    # Check if first index is wrapped (decreasing coordinate).
+                    # i.e. start of the second chunk is at the second index.
+                    elif index_of_second_chunk == 1:
+                        # Unwrap points and bounds
+                        if (bounds is not None and
+                                np.isclose(coord.bounds[0, 0],
+                                           coord.bounds[-1, -1] + modulus)):
+                            bounds[0, :] = coord.bounds[0, :]
+                            points[0] = coord.points[0]
+                            subsets = [slice(inside_indices[0],
+                                             inside_indices[-1] + 1)]
+                        # Unwrap points
+                        elif np.isclose(coord.points[0],
+                                        coord.points[-1] + modulus):
+                            points[0] = coord.points[0]
+                            subsets = [slice(inside_indices[0],
+                                             inside_indices[-1] + 1)]
+                else:
+                    subsets = [
+                        slice(index_of_second_chunk, None),
```

Comment:
This line changed during your development but doesn't seem to have been any corresponding change in the tests - perhaps it hasn't been covered by the tests? Or perhaps the tests didn't always pass?